src/stcal/jump/jump.py

```diff
@@ -54,6 +54,8 @@ def detect_jumps(
     minimum_groups=3,
     minimum_sigclip_groups=100,
     only_use_ints=True,
+    mask_persist_grps_next_int=True,
+    persist_grps_flagged=25,
```

Comment:
This function now has roughly 30 arguments.  Would it be better served by using a class to contain the parameters, rather than continue to add parameters to this function?

Reply:
I agree that it has a lot of parameters but users need to be able to tweak the parameters. Is there any cost to the large number of parameters?

Reply:
The cost is readability and maintainability.  Grouping various parameters into a class or classes makes the code easier to read, as well as easier to maintain, especially when adding a parameter that needs to be passed to subroutines.  If you simply pass parameters, the function signature needs to be changed for all functions that need that parameter.  If you use classes, those classes will be passed to the subroutines, so when you add a parameter to a class, no subroutine signature needs to be edited, since the parameter is contained in the class.

Reply:
While it is true that passing objects instead of primitives  has advantages there are also disadvantages. APIs with only primitive parameters are easier to understand where the primitives are coming from. At the extreme, a function that has a single class as input in a black box. Someone browsing the code has more work to do than with primitive parameters. 
  Also, If we want to consolidate the parameters passed to detect_jump.py, I feel that the change should be in a separate PR. That would give us a working master to compare against. 
  What do you think?